cmd/kwil-cli/cmds/common/roundtripper.go

```diff
@@ -54,18 +55,23 @@ func DialClient(ctx context.Context, cmd *cobra.Command, flags uint8, fn RoundTr
 		if needPrivateKey { // only check chain ID if signing something
 			clientConfig.ChainID = conf.ChainID
 		}
-	} else if flags&WithoutPrivateKey == 0 {
+	} else if flags&WithoutPrivateKey == 0 && flags&AuthenticatedCalls == 0 {
```

Comment:
Bitmask logic plus double negatives is dizzing.  Let's do above:

```go
needPrivateKey := flags&WithoutPrivateKey == 0
authCalls := flags&AuthenticatedCalls != 0
```

and then write all the conditions in this function in terms of these local variables.

I'm pretty confused about this current condition.